### PR TITLE
fake_sensor_commands renamed

### PR DESCRIPTION
**Pull request by kvkpraneeth**
This PR resolves https://github.com/ros-controls/ros2_control/issues/774 issue to refactor all of `fake_sensor_commands` to `mock_sensor_commands`, Also adds a small deprecation warning if `fake_sensor_commands` is used.

### CODE REVIEW

**Review by mcbed — DISMISSED** [review]
Looks good to me

File: hardware_interface/src/mock_components/generic_system.cpp [hunk 1]
```diff
@@ -57,16 +57,29 @@ CallbackReturn GenericSystem::on_init(const hardware_interface::HardwareInfo & i
     }
   };
 
-  // check if to create fake command interface for sensor
-  auto it = info_.hardware_parameters.find("fake_sensor_commands");
```
Comment (destogl) [inline hardware_interface/src/mock_components/generic_system.cpp:61]: @bmagyar should be deprecate this first?

**Review by destogl — DISMISSED** [review]
Thanks!

File: hardware_interface/src/mock_components/generic_system.cpp (hunk 1 above)
Comment (bmagyar) [reply hardware_interface/src/mock_components/generic_system.cpp:61]: below it is still processed, the deprecation is in place IMO.

Can you confirm that is the intent @kvkpraneeth ?

Comment (kvkpraneeth) [reply hardware_interface/src/mock_components/generic_system.cpp:61]: Yes, that was the intent

**Comment by destogl** [comment]
@bmagyar if deprecation is in place we should merge this.

File: hardware_interface/test/mock_components/test_generic_system.cpp [hunk 2]
```diff
@@ -215,7 +215,7 @@ class TestGenericSystem : public ::testing::Test
   <ros2_control name="GenericSystem2dof" type="system">
     <hardware>
       <plugin>mock_components/GenericSystem</plugin>
-      <param name="fake_sensor_commands">True</param>
+      <param name="mock_sensor_commands">True</param>
```
Comment (bmagyar) [inline hardware_interface/test/mock_components/test_generic_system.cpp:218]:
```suggestion
      <param name="fake_sensor_commands">True</param>
```

Comment (bmagyar) [reply hardware_interface/test/mock_components/test_generic_system.cpp:218]: Let's leave one as-is to test the deprecation

**Review by bmagyar — APPROVED** [review]

**Review by destogl — DISMISSED** [review]

**Comment by destogl** [comment]
@mergifyio backport humble

**Comment by destogl** [comment]
@mergifyio backport humble

**Pull request merged**